test/terra/states/test_aer_statevector.py

```diff
@@ -0,0 +1,112 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2019, 2020, 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Integration Tests for Parameterized Qobj execution, testing qasm_simulator,
+statevector_simulator, and expectation value snapshots.
+"""
+
+import unittest
+from math import pi
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.quantum_info.random import random_unitary
+from qiskit.quantum_info.states import Statevector
+from qiskit.circuit.library import QuantumVolume
+from qiskit.providers.aer import AerSimulator
+
+from test.terra import common
+from qiskit.providers.aer.aererror import AerError
+
+class TestAerStatevector(common.QiskitAerTestCase):
```

Comment:
I think it might be a good idea if we copied more from https://github.com/Qiskit/qiskit-terra/blob/main/test/python/quantum_info/states/test_statevector.py and update it to use aer statevector so we can test full compatibility (or as much as we can get)